services/skus/controllers.go

```diff
@@ -52,17 +52,42 @@ func Router(service *Service, instrumentHandler middleware.InstrumentHandlerDef)
 	orderh := handler.NewOrder(service)
 
 	if os.Getenv("ENV") == "local" {
-		r.Method(http.MethodOptions, "/", middleware.InstrumentHandler(
-			"CreateOrderOptions",
-			corsMiddleware([]string{http.MethodPost})(nil),
-		))
-
-		r.Method(http.MethodPost, "/", middleware.InstrumentHandler(
-			"CreateOrder",
-			corsMiddleware([]string{http.MethodPost})(handlers.AppHandler(orderh.Create)),
-		))
+		r.Method(
+			http.MethodOptions,
+			"/",
+			middleware.InstrumentHandler("CreateOrderOptions", corsMiddleware([]string{http.MethodPost})(nil)),
+		)
+
+		r.Method(
+			http.MethodOptions,
+			"/new",
+			middleware.InstrumentHandler("CreateOrderNewOptions", corsMiddleware([]string{http.MethodPost})(nil)),
+		)
+
+		r.Method(
+			http.MethodPost,
+			"/",
+			middleware.InstrumentHandler(
+				"CreateOrder",
+				corsMiddleware([]string{http.MethodPost})(handlers.AppHandler(orderh.Create)),
+			),
+		)
+
+		r.Method(
+			http.MethodPost,
+			"/new",
+			middleware.InstrumentHandler(
+				"CreateOrderNew",
+				corsMiddleware([]string{http.MethodPost})(handlers.AppHandler(orderh.CreateNew)),
+			),
+		)
 	} else {
 		r.Method(http.MethodPost, "/", middleware.InstrumentHandler("CreateOrder", handlers.AppHandler(orderh.Create)))
+		r.Method(
+			http.MethodPost,
+			"/new",
```

Comment:
As we discussed this IRL, no, we are not following RESTful principles here, unfortunately. 😐

In strict terms, yes, it will break the convention. The available alternatives (adding a `/v2/orders` or branching out inside the existing endpoint and handler) do not sound attractive for a number of reasons.

We need a new endpoint, but it's not enough for a `v2`, and we can't simply reuse the existing one because it's open.

So far my two ideas are:
- Either do this, which is verging on to the RPC territory, – `/v1/orders/new`
- Or do something similar, but ultimately cleaner – `/v1/orders-new`.

Now as I am looking at this, I like the second option more. Still ugly, but at least does not break the convention.

Let me know what you think @clD11.

Reply:
yes, the second option seem the lesser of the two evils :laughing:

Reply:
Working on this. Hoping to make it a tiny bit better (the way routing is currently initialised).

Reply:
@clD11 When convenient, please take a look at #1938. There, I have slightly improved the situation with the router initialisation in SKUs, and changed code from this PR to use `/v1/orders-new`.

The base branch for that PR is this branch, so if/once you're happy with it, I will merge that in here. Thought it would be easier to look at the changes in some isolation.